### PR TITLE
Bug fix: RangeInput onChange

### PR DESCRIPTION
**Pull request by StheffanyHadlich**
#### What does this PR do?

This PR solves #4275 adding a type to `RangeInput onChange`

### CODE REVIEW

File: src/js/components/RangeInput/index.d.ts [hunk 1]
```diff
@@ -7,7 +7,7 @@ export interface RangeInputProps {
   min?: number | string;
   max?: number | string;
   name?: string;
-  onChange?: ((event: React.ChangeEvent) => void);
+  onChange?: ((event: React.ChangeEvent<HTMLInputElement>) => void);
```
Comment (ShimiSun) [inline src/js/components/RangeInput/index.d.ts:10]: Thank you for this PR! 
Can you please add a few words why you think this fixes the problem?

**Comment by ShimiSun** [comment]
@carolineartz I'd love your insight here.

**Comment by carolineartz** [comment]
This will fix the problem...but I'm wondering why we're re-defining the types at all.... the `onChange` for an input has this type already defined internally, with the properly typed generic.  On first glance all of the props in `RangeProps` are already on `input` element and the `RangeInput` only wraps `input` to intercept the values for `formContext` -- doesn't add anything custom. 

Would this just be fixed by getting rid of `RangeInputProps` all together and instead:

```ts
declare const RangeInput: React.FC<JSX.IntrinsicElements['input']>;

const RangeInput = forwardRef<HTMLInputElement, JSX.IntrinsicElements['input']>(
  ({ name, onChange, onFocus, onBlur, value: valueProp, ...rest }, ref) => {
  ...
```

cc @ShimiSun

**Comment by ShimiSun** [comment]
How will this approach work with props that are not part of the input IntrinsicElements? Such as `a11yTitle` prop?

File: src/js/components/RangeInput/index.d.ts (hunk 1 above)
Comment (taysea) [reply src/js/components/RangeInput/index.d.ts:10]: Found a resource here that seems to mirror this solution: https://blaipratdesaba.com/react-typescript-cheatsheet-form-elements-and-onchange-event-types-8c2baf03230c

**Comment by taysea** [comment]
> @carolineartz, How will this approach work with props that are not part of the input IntrinsicElements? Such as `a11yTitle` prop?

When I comment out the `onChange` from RangeInputProps, I do see the intrinsic property and it matches the proposed fix. Since we are not making any modifications to the handler should we remove it from the interface? or is the convention that we want to display all handlers regardless of if we are modifying them or not (if this is the case we can move forward with the proposed change).

**Comment by ShimiSun** [comment]
@taysea good question, since we are not making any modifications on the handler, we should fall back to its default intrinsic definition, so unless we are adding more functionalities to it, I think we can remove it.

**Comment by taysea** [comment]
> @taysea good question, since we are not making any modifications on the handler, we should fall back to its default intrinsic definition, so unless we are adding more functionalities to it, I think we can remove it.

Great, I've opened https://github.com/grommet/grommet/pull/4429 which implements these changes.

**Comment by ShimiSun** [comment]
Thank you @carolineartz, @taysea & @StheffanyHadlich for working on this PR and getting it to the finish line!

**Pull request closed without merging**